components/notebook-controller/config/overlays/service-mesh/gateway.yaml

```diff
@@ -0,0 +1,18 @@
+apiVersion: networking.istio.io/v1beta1
+kind: Gateway
+metadata:
+  name: odh-gateway
+  namespace: odh-notebook-controller-system
+spec:
+  selector:
+    istio: ingressgateway
+  servers:
+  - port:
+      number: 443
+      name: https
+      protocol: HTTPS
+    tls:
+      mode: SIMPLE
+      credentialName: odh-dashboard-cert
```

Comment:
If the overlays here are used mainly for testing purposes I can adjust tests to be using `http` if desired.